php_phongo.h

```diff
@@ -60,28 +60,112 @@ ZEND_TSRMLS_CACHE_EXTERN()
 
 zend_object_handlers* phongo_get_std_object_handlers(void);
 
+#define PHONGO_RETURN_PROPS(is_temp, props) \
+	if (!(is_temp)) { \
+		GC_ADDREF(props); \
```

Comment:
This is done to mark that the hashtable is also stored in (intern)->properties; can't be done in `PHONGO_GET_PROPERTY_HASH_INIT_PROPS` because `zend_hash_str_update` to add the native properties requires the refcount of the hashtable to be 1.